packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts

```diff
@@ -104,8 +111,35 @@ export default class SchedulePaymentFormActionCard extends Component<Signature>
   }
 
   @action
-  schedulePayment() {
-    console.log('TODO...');
+  async schedulePayment() {
+    let { currentSafe } = this.safes;
+    if (!currentSafe) return;
+    if (!this.paymentDate) return;
+    if (!this.paymentToken) return;
+    if (!this.selectedGasToken) return;
+
+    const array = new Uint8Array(32);
+    window.crypto.getRandomValues(array);
+    const salt = btoa(String.fromCharCode.apply(null, array));
+
+    await taskFor(this.scheduledPaymentsSdk.schedulePayment).perform(
+      currentSafe.address,
+      currentSafe.spModuleAddress,
+      this.paymentToken.address,
+      convertAmountToRawAmount(this.paymentAmount, this.paymentToken.decimals),
+      this.payeeAddress,
+      this.executionGas,
+      '15000000000', // TODO: this.maxGasFee,
```

Comment:
Would be good to use `maxGasPrice` name so that it is consistent with the name in the SDK and the contract.